lib/src/deserializer.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// deserializer.dart
+// A deserialization block, deserializing narrow input data onto a wide channel.
+//
+// 2024 August 27
+// Author: desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// Aggregates data from a serialized stream
+class Deserializer extends Module {
+  /// Clk input
+  @protected
+  Logic get clk => input('clk');
+
+  /// Reset input
+  @protected
+  Logic get reset => input('reset');
+
+  /// Run deserialization whenever [enable] is true
+  @protected
+  Logic? get enable => input('enable');
+
+  /// Serialized input, one data item per clock
+  Logic get serialized => input('serialized');
```

Comment:
this should be `@protected`

Reply:
All inputs should be protected?

Reply:
This is an input.  But you mention @protected for outputs.  I'm not clear on what we want to do here.